src/libs/logging/src/writer.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2020 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::io::{Result, Write};
+
+use slog::{info, Logger};
+
+/// Writer to convert each line written to it to a log record.
+#[derive(Debug)]
+pub struct LogWriter(Logger);
```

Comment:
AFAICT this isn't really related to `FileRotator` except in how it is tested.  It should be mentioned in the commit message.

Reply:
Will split it into a dedicated commit.